### PR TITLE
[ARM] `az deployment`: Fix the bug of `'bytes' object has no attribute 'get'` for error handling in retry cases

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Issue: #19743

**Description**

This error `bytes' object has no attribute 'get'` is due to the bug of `JsonCTemplatePolicy` in the retry operation. 
In the case of retry, because the first request has been processed and converted the type of `request.http_request.data` from string to bytes [code link](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/custom.py#L425)  , so there is no need to process `request.http_request.data` again during retry, 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -410,7 +410,15 @@ class JsonCTemplatePolicy(SansIOHTTPPolicy):
 
     def on_request(self, request):
         http_request = request.http_request
-        if (getattr(http_request, 'data', {}) or {}).get('properties', {}).get('template'):
+        request_data = getattr(http_request, 'data', {}) or {}
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:413]: @zhoxing-ms, have we run all tests impacted by this logic?

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py:413]: It mainly runs the tests that affects the scope of resource group. The logic of other scopes (such as: tenant sub mg) calling this policy is the same, so the scope of resource group can basically cover all use cases~

**Review by jsntcy — APPROVED** [review]

**Comment by jiasli** [comment]
The logic of `JsonCTemplatePolicy` is hard to understand. We need better comments!

**Comment by zhoxing-ms** [comment]
> It seems JSONSerializer wraps the jsonc in a JsonCTemplate and then JsonCTemplatePolicy extracts the jsonc. Why isn't the jsonc serialization logic done in JSONSerializer?

In fact, it's technically feasible. I've tried it and it works fine. I guess this may be a decision in code design. Maybe this splicing is not appropriate to put it in the serialization logic in code design.
Because this is legacy code, I can't answer this question perfectly. 😂

> Also, The name template_as_bytes is misleading (#10389 (comment)).

I agree! It will be converted to bytes in the subsequent step of `JsonCTemplatePolicy`, but not in the step of `JSONSerializer`. So I submitted a PR #21991 to correct the variable name and add some comments to explain the function of `JsonCTemplatePolicy`

**Pull request merged**